src/otx/algo/utils/xai_utils.py

```diff
@@ -11,44 +11,96 @@
 from otx.core.config.explain import ExplainConfig
 from otx.core.data.entity.base import OTXBatchPredEntityWithXAI
 from otx.core.data.entity.instance_segmentation import InstanceSegBatchPredEntityWithXAI
+from otx.core.types.explain import TargetExplainGroup
 
 if TYPE_CHECKING:
     from pathlib import Path
 
+    import numpy as np
 
-def get_processed_saliency_maps(
-    predictions: list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI] | None,
-    explain_config: ExplainConfig,
-    work_dir: Path | None,
-) -> list:
-    """Implement saliency map filtering and post-processing."""
-    # Optimize for memory <- TODO(negvet)
-    raw_saliency_maps: list = []
-    if predictions:
-        raw_saliency_maps = predictions[0].saliency_maps
 
-    if raw_saliency_maps and work_dir:
-        # Temporary saving saliency map for image 0, class 0 (for tests)
-        cv2.imwrite(str(work_dir / "saliency_map.tiff"), raw_saliency_maps[0][0])
+def process_saliency_maps_in_pred_entity(
+    predict_result: list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI],
+    explain_config: ExplainConfig,
+    work_dir: Path | None = None,
+) -> list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI]:
+    """Process saliency maps in PredEntity."""
+    saliency_maps = predict_result[0].saliency_maps
```

Comment:
`predict_result` is supposed to be a list of per-batch saliency maps. Does this line mean we process only the first batch?

Reply:
Right, it was not supported. I fixed it.